kafka-client/src/main/java/dev/responsive/kafka/internal/db/rs3/RS3KVTable.java

```diff
@@ -0,0 +1,139 @@
+package dev.responsive.kafka.internal.db.rs3;
+
+import dev.responsive.kafka.internal.db.KVFlushManager;
+import dev.responsive.kafka.internal.db.RemoteKVTable;
+import dev.responsive.kafka.internal.db.rs3.client.LssId;
+import dev.responsive.kafka.internal.db.rs3.client.Put;
+import dev.responsive.kafka.internal.db.rs3.client.RS3Client;
+import dev.responsive.kafka.internal.db.rs3.client.WalEntry;
+import dev.responsive.kafka.internal.stores.ResponsiveStoreRegistration;
+import java.util.HashMap;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.UUID;
+import org.apache.kafka.common.utils.Bytes;
+import org.apache.kafka.streams.state.KeyValueIterator;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class RS3KVTable implements RemoteKVTable<WalEntry> {
+  private static final Logger LOG = LoggerFactory.getLogger(RS3KVTable.class);
+
+  private final String name;
+  private final UUID storeId;
+  private final RS3Client rs3Client;
+  private final PssPartitioner pssPartitioner;
+  private LssId lssId;
+  private Long fetchOffset = ResponsiveStoreRegistration.NO_COMMITTED_OFFSET;
+  private RS3KVFlushManager flushManager;
+
+  public RS3KVTable(
+      final String name,
+      final UUID storeId,
+      final RS3Client rs3Client,
+      final PssPartitioner pssPartitioner
+  ) {
+    this.name = Objects.requireNonNull(name);
+    this.storeId = Objects.requireNonNull(storeId);
+    this.rs3Client = Objects.requireNonNull(rs3Client);
+    this.pssPartitioner = Objects.requireNonNull(pssPartitioner);
+  }
+
+  @Override
+  public KVFlushManager init(final int kafkaPartition) {
+    if (flushManager != null) {
+      throw new IllegalStateException("already initialized");
+    }
+
+    this.lssId = new LssId(kafkaPartition);
+
+    // TODO: we should write an empty segment periodically to any PSS that we haven't
+    //       written to to bump the written offset
+    final HashMap<Integer, Optional<Long>> lastWrittenOffset = new HashMap<>();
```

Comment:
might be nice to have a strongly typed `Pss` class just for readability throughout the code even if it just wraps an integer (though it could encapsulate some of the bytestring prefix stuff too)